e2e/output_runner/output_runner_test.go

```diff
@@ -69,9 +71,10 @@ func checkNoSentinel(t *testing.T, sentinelFile *os.File, msg string) {
 func checkSentinel(t *testing.T, sentinelFile *os.File, msg string) {
 	t.Helper()
 
+	time.Sleep(500 * time.Millisecond)
```

Comment:
This `time.Sleep` totally makes sense to me because the process is supposed to be waiting for something to happen and the test is parellized. WDYT about turning this into a fast path exit:

```
deadline := time.Now().Add(5*time.Second)
var err error
for {
  if time.Now().After(deadline) {
    t.Errorf("Couldn't find sentinal. os.Stat(%q): %s\n%s\n", sentinalFileName, err, msg)
    return
  } else if _, err := os.Stat(sentinalFileName); err == nil {
    // No error stat'ing the file means it exists.
    return
  }
}
```

Now it'll tight loop until it succeeds or timeout happens.

Reply:
Yea thats a good approach, thanks for the suggestion